stock_picking_group_by_partner_by_carrier/models/stock_picking.py

```diff
@@ -139,6 +139,15 @@ def get_delivery_report_lines(self):
             else:
                 return self.move_line_ids
 
+    def get_customer_refs(self):
+        """Returns all unique sales order customer references."""
+        if self.state != "done":
+            move_lines = self.move_lines.filtered("product_uom_qty")
+        else:
+            move_lines = self.move_lines
```

Comment:
Not sure to understand this trick?
There are moves without `product_uom_qty` filled? What is the link with the state of the picking?